prow/genjobs/go.mod

```diff
@@ -0,0 +1,14 @@
+module istio.io/test-infra/prow/genjobs
```

Comment:
I'm concerned that we have so many submodules in this repo. Why do we need/want them?

https://github.com/golang/go/wiki/Modules#should-i-have-multiple-modules-in-a-single-repository

Reply:
Originally this was intended to resolve the `go get ...` issue and any k8s dependency conflicts with the root `go.mod`. 

Since this uses more-or-less the same deps as the root (i.e. no reduced image size benefit) and does not solve the `go get ...` dilemma (i.e. b/c it depends on `k8s/test-infra` directly (e.g. https://github.com/kubernetes/test-infra/issues/14070)) maybe submodule for this does **not** make sense after all.

Reply:
Conversely, for tools like `authentikos`, I think there is a clear benefit to using submodules for modularity, image size reduction, and usability.